### PR TITLE
[#339] Replace dependency to org.eclipse.osgi.services

### LINKED ISSUE #339: Replace dependency to org.eclipse.osgi.services
The bundle `o.e.osgi.services` is deprecated for removal as part of https://github.com/eclipse-equinox/equinox/issues/18 and already has been changed to be empty and only re-export the 'official' OSGi artifacts published to Maven-Central.
It should be replaced by imports of the desired `org.osgi.service.*` packages so that the runtime can choose a suitable provider independently of the bundle name.

### PR DESCRIPTION
**Pull request by ruspl-afed**
Move to
```
Import-Package: org.osgi.service.event;version="1.4.1"
```

Fixed #339

### CODE REVIEW

File: mylyn.builds/org.eclipse.mylyn.builds.ui/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -57,4 +56,5 @@ Export-Package: org.eclipse.mylyn.builds.ui;x-internal:=true,
  org.eclipse.mylyn.internal.builds.ui.notifications;x-internal:=true,
  org.eclipse.mylyn.internal.builds.ui.util;x-internal:=true,
  org.eclipse.mylyn.internal.builds.ui.view;x-internal:=true
+Import-Package: org.osgi.service.event;version="1.4.1"
```
Comment (HannesWell) [inline mylyn.builds/org.eclipse.mylyn.builds.ui/META-INF/MANIFEST.MF:59]: In general it is good practice to specify an upper bound to follow semantic versioning.
```suggestion
Import-Package: org.osgi.service.event;version="[1.4.0,2.0.0)"
```

**Review by HannesWell — COMMENTED** [review]
Thanks for working on this so quickly.
The affected bundles match the analysis done by Ed using the CBI Analyser, so that looks good.
But I would specify an upper bound for the imports.

File: mylyn.builds/org.eclipse.mylyn.builds.ui/META-INF/MANIFEST.MF (hunk 1 above)
Comment (ruspl-afed) [reply mylyn.builds/org.eclipse.mylyn.builds.ui/META-INF/MANIFEST.MF:59]: Might be good improvement for PDE :)

Comment (HannesWell) [reply mylyn.builds/org.eclipse.mylyn.builds.ui/META-INF/MANIFEST.MF:59]: We are working on it :)
https://github.com/eclipse-pde/eclipse.pde/pull/1007#pullrequestreview-1782918647

Comment (ruspl-afed) [reply mylyn.builds/org.eclipse.mylyn.builds.ui/META-INF/MANIFEST.MF]: Great! I have improved the constraints according to your recommendations.

**Review by HannesWell — APPROVED** [review]
Looks good to me.
Just one tiny nit-pick, the micro version segment can usually be zero as well.

**Pull request merged**